vm/src/builtins/function.rs

```diff
@@ -406,6 +416,41 @@ impl PyFunction {
         *self.name.lock() = name;
     }
 
+    #[pymember(magic)]
+    fn doc(_vm: &VirtualMachine, zelf: PyObjectRef) -> PyResult {
+        let zelf: PyRef<PyFunction> = zelf.downcast().unwrap_or_else(|_| unreachable!());
+        let doc = zelf.doc.lock();
+        Ok(doc.clone())
+    }
+
+    #[pymember(magic, setter)]
+    fn set_doc(vm: &VirtualMachine, zelf: PyObjectRef, value: PySetterValue) -> PyResult<()> {
+        let zelf: PyRef<PyFunction> = zelf.downcast().unwrap_or_else(|_| unreachable!());
+        let value = value.unwrap_or_none(vm);
+        *zelf.doc.lock() = value;
+        Ok(())
+    }
+
+    #[pygetset(magic)]
+    fn module(&self) -> PyObjectRef {
+        self.module.lock().clone()
+    }
+
+    #[pygetset(magic, setter)]
+    fn set_module(&self, module: PyObjectRef) {
```

Comment:
It's going well! `__module__` need to support delete operation

```suggestion
    fn set_module(&self, module: PySetterValue) {
```
To support deletion, the value type need to be changed. Using `match` statement helps to distinguish set and delete operation

CI Result:
```
======================================================================
ERROR: test_local_lookup_error (test.test_pickle.InMemoryPickleTests.test_local_lookup_error)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/RustPython/RustPython/pylib/Lib/test/pickletester.py", line 2885, in test_local_lookup_error
    del f.__module__
TypeError: can't delete attribute

======================================================================
ERROR: test_local_lookup_error (test.test_pickle.PyPicklerTests.test_local_lookup_error)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/RustPython/RustPython/pylib/Lib/test/pickletester.py", line 2885, in test_local_lookup_error
    del f.__module__
TypeError: can't delete attribute
```

Reply:
Looks like the `__module__` is `None` after deletion in CPython, so i implemented it that way